app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt

```diff
@@ -59,29 +59,32 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
         SensorReceiver.MANAGERS.sortedBy { it.name }.forEach { manager ->
             val prefCategory = PreferenceCategory(preferenceScreen.context)
             prefCategory.title = getString(manager.name)
-            preferenceScreen.addPreference(prefCategory)
-            manager.availableSensors.sortedBy { it.name }.forEach { basicSensor ->
 
-                val pref = Preference(preferenceScreen.context)
-                pref.key = basicSensor.id
-                pref.title = getString(basicSensor.name)
+            if (manager.hasSensor(requireContext())) {
```

Comment:
If you really wanted to get fancy you could filter before the foreach and remove the whole if statement.